packages/@lwc/engine-core/src/framework/component.ts

```diff
@@ -33,6 +34,11 @@ export function registerComponent(
             checkVersionMismatch(Ctor, 'component');
         }
         signedTemplateMap.set(Ctor, tmpl);
+
+        // Do not assign values for empty strings.
+        if (sel) {
+            registeredComponentNameMap.set(Ctor, sel);
+        }
```

Comment:
Should we throw an error here if `sel` is missing?

Reply:
I planned to make this throw an error in a future update once the feature has been tested further.  I want to make sure I'm not missing any edge cases.  

There is a [check](https://github.com/salesforce/lwc/pull/3338/files#diff-a284e53a17a2b48cc32e6bf3b121b3acb849a57df886d1cb92a2a46df5aebb7bR595-R599) that verifies the constructor has a registered name and throws an error otherwise.

Reply:
Yup, I added a [comment](https://github.com/salesforce/lwc/pull/3338/commits/b7dcd96d43735e82762883ae19b7b9057bce2479#diff-0550a116baa4d41be80b364dde9e93891ab20453650053580f15044e9e46d2b3R40) in the latest commit